lib/msf/core/exploit/remote/http/nagios_xi/login.rb

```diff
@@ -52,30 +53,37 @@ def nagios_xi_login(user, pass)
       return Msf::Exploit::CheckCode::Detected('Received unexpected reply while trying to authenticate')
     end
 
-    # keep only the most recent cookie, otherwise the session will time out
-    auth_cookies = cookie_jar.to_a[1]
-    self.cookie_jar = auth_cookies.split.to_set
+    # grab the cookies
+    auth_cookies = res_login.get_cookies
+
+    if auth_cookies.blank?
+      return Msf::Exploit::CheckCode::Unknown('Unable to obtain the cookies required for authentication')
```

Comment:
Presumably this is due to authentication failure, in which case `Failure::NoAccess` is more accurate.